### ISSUE TITLE
Creating mask on the fly with limited pixel information

### ISSUE THREAD

**Issue by jamesbc123** [open-mmlab/mmdetection#7840]
I want to create a mask that contains a limited number of pixels in it (out of the original mask e.g. from COCO dataset). And I want to do this on the fly. Would anyone know where I can edit the mask so that e.g. only the centre pixel of the original mask is used for training, and the rest can be a value that the loss function ignores. Should I create a function to do this in mmdet/core/mask/structures.py or maybe in datasets/pipelines/transforms.py? Any alternatives?

Thank you!

**Comment by jbwang1997** [comment]
In your case, you can create a new pipeline to process the original masks. You can set a score to each pixel (e.g. the distance between this pixel and the object center) and get the top k pixel's score. By applying `score > topk_score`, you can limit the pixel number in masks.

**Comment by jamesbc123** [comment]
Thanks so much!

For the loss function since I now have lots of pixels to ignore (I gave them value -1) then I want to use cross entropy not binary cross entropy. Binary cross entropy doesn't allow ignore index. But the only way to do that is to set use_mask=False in the config file under mask_head. Do I need to create my own loss function then... to use ignore index with a mask?
